src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptor.java

```diff
@@ -265,11 +265,19 @@
 
     @Override public Object onGetProperty(GroovyInterceptor.Invoker invoker, final Object receiver, final String property) throws Throwable {
         MissingPropertyException mpe = null;
-        if (receiver instanceof Script) { // SimpleTemplateEngine "out" variable, and anything else added in a binding
+        if (receiver instanceof Script) { // SimpleTemplateEngine "out" variable, anything else added in a binding, and @Field variables.
             try {
                 ((Script) receiver).getBinding().getVariable(property); // do not let it go to Script.super.getProperty
                 return super.onGetProperty(invoker, receiver, property);
             } catch (MissingPropertyException x) {
+                if (receiver.getClass() != Script.class) { // Only check declared fields of subclasses, not Script itself
```

Comment:
Maybe unnecessary, since binding is the only field declared directly in `Script`.